httputil/cert_ext_tripper.go

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package httputil
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/asn1"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/hashicorp/go-hclog"
+)
+
+type ignoreExtensionsRoundTripper struct {
+	base         *http.Transport
+	extsToIgnore []asn1.ObjectIdentifier
+	logger       hclog.Logger
+}
+
+// NewIgnoreUnhandledExtensionsRoundTripper creates a RoundTripper that may be used in an HTTP client which will
+// ignore the provided extensions if present unhandled on a certificate.  If base is nil, the default RoundTripper is used.
+func NewIgnoreUnhandledExtensionsRoundTripper(logger hclog.Logger, base http.RoundTripper, extsToIgnore []asn1.ObjectIdentifier) http.RoundTripper {
+	if len(extsToIgnore) == 0 {
+		return base
+	}
+	if base == nil {
+		base = http.DefaultTransport
+	}
+
```

Comment:
This function is missing a check that the transport's TLSClientConfig is not nil, and erroring if it is. Right now it appears that if a non-TLS transport is provided, including using the DefaultTransport if it has been given no TLS info, you'll panic in RoundTrip.

Reply:
That happens later, we initialize a TLSConfig if missing in RoundTrip.  This is to avoid needing to modify the base RoundTripper.

Reply:
Ah right, I missed the base clone, thanks.

Reply:
Ah, sorry, I mixed up that if RoundTrip is missing `http.DefaultTransport` will use a system TLS config. 👍